src/Microsoft.AspNetCore.Http.Abstractions/ConnectionInfo.cs

```diff
@@ -10,6 +10,11 @@ namespace Microsoft.AspNetCore.Http
 {
     public abstract class ConnectionInfo
     {
+        /// <summary>
+        /// Gets or sets a unique identifier to represent this connection in trace logs.
+        /// </summary>
+        public abstract string TraceIdentifier { get; set; }
```

Comment:
:-1: this is mixing names. We already have a TraceIdentifier on HttpContext with a different source value (actually made for tracing). Just call it ConnectionId